Fix duration parsing error handling

vpaturet opened the pull request:
### Summary

A request that specifies a time penalty with an incorrect duration format:
```
accessEgressPenalty: [{
      streetMode: flexible
      timePenalty: "600ss + 1.3 t"
      costFactor: 1.3
    }],

```
fails with an HTTP 500 error:

```

"Text cannot be parsed to a Duration: '600ss' java.time.format.DateTimeParseException: Text cannot be parsed to a Duration: '600ss'
	at org.opentripplanner.framework.time.DurationUtils.duration(DurationUtils.java:121)
	at org.opentripplanner.routing.api.request.framework.LinearFunctionSerialization.parse(LinearFunctionSerialization.java:68)
	at org.opentripplanner.apis.transmodel.model.scalars.DoubleFunctionFactory$1.parseValue(DoubleFunctionFactory.java:59)
	at org.opentripplanner.apis.transmodel.model.scalars.DoubleFunctionFactory$1.parseLiteral(DoubleFunctionFactory.java:73)
	at org.opentripplanner.apis.transmodel.model.scalars.DoubleFunctionFactory$1.parseLiteral(DoubleFunctionFactory.java:40)

```

This PR fixes exception handling so that the parsing error is reported as a client error:
```

{
  "errors": [
    {
      "message": "Unable to parse duration: '600ss'",
      "locations": [],
      "extensions": {
        "classification": "BadRequestError"
      }
    }
  ]
}
```

**Note**: the parsing logic is moved to a private method `LinearFunctionSerialization.parseDecimalSecondsOrDuration()`. This method should probably not be promoted to the class `DurationUtils`: it is used solely to fix a backward compatibility issue (see #5304).

### Issue

No

### Documentation

No

leonardehrenfried reviewed (changes requested):
Is it possible to write a test for this behaviour?

vpaturet commented:
I've added a couple of test cases to test the behavior with decimal seconds.
I see now that the corner case with invalid duration was not covered, so I added a test case for this s well.
